[BUG] Missing scripts `npm run build:all` | `npm run publish:all`

# 🐞 Bug report

<!-- A clear and concise description of what the bug is -->

### Reproduction
Run this script
https://github.com/Tinkoff/angular-open-source-starter/blob/a6e70777fcfd05d98959a419d9deba689ba27d14/package.json#L34

it throws error:
```console
➜  angular-open-source-starter git:(test) ✗ npm run publish      

> @tinkoff/angular-open-source-starter@0.0.1 publish
> npm run build:all && npm run publish:all

npm ERR! Missing script: "build:all"

```

### Expected behavior
No error :)

--- a/README.md
+++ b/README.md
@@ -64,8 +64,9 @@ automatically generated on releases by
 
          https://stackblitz.com/github/[User|Organization]/[Repository]/tree/main/projects/demo
 
-- [x] You can add more libraries using the same `npm run add` command to create a whole Angular Workspace with multiple
-      libraries. Versioning and publishing is configured that they are released simultaneously like Angular packages.
+- [x] You can add more libraries using the same [Nx-generators](https://nx.dev/plugin-features/use-code-generators) to
+      create a whole Angular Workspace with multiple libraries. Versioning and publishing is configured that they are
+      released simultaneously like Angular packages.
 
 ## Infrastructure
 

--- a/README.md
+++ b/README.md
@@ -42,7 +42,7 @@
   "release:patch": "npm run release -- --release-as patch",
   "release:minor": "npm run release -- --release-as minor",
   "release:major": "npm run release -- --release-as major",
-  "publish": "npm run build:all && npm publish:all"
+  "publish": "nx run-many --target build --all --exclude=demo && nx run-many --target publish --all"
 }
 ```
 

--- a/package.json
+++ b/package.json
@@ -26,7 +26,7 @@
         "release:patch": "npm run release -- --release-as patch",
         "release:minor": "npm run release -- --release-as minor",
         "release:major": "npm run release -- --release-as major",
-        "publish": "npm run build:all && npm run publish:all"
+        "publish": "nx run-many --target build --all --exclude=demo && nx run-many --target publish --all"
     },
     "description": "This is a template project to develop, create and open-source Angular libraries with demos",
     "keywords": [

--- a/scripts/helpers/argv.ts
+++ b/scripts/helpers/argv.ts
@@ -1,0 +1,34 @@
+import * as process from 'process';
+
+import {processLog} from './colored-log';
+
+export function getValueByFlag<T extends string>(flag: string, fallback: T): T {
+    const index = findIndexFlag(flag);
+
+    if (index === -1) {
+        return fallback;
+    }
+
+    const [parsedFlag, parsedValue] = process.argv[index].split(`=`) ?? [];
+    const value =
+        stringifier(parsedValue) ??
+        (process.argv[index + 1].startsWith(`-`)
+            ? fallback
+            : stringifier(process.argv[index + 1]) ?? fallback);
+
+    processLog(`parsed flags: \n${[parsedFlag, value].join(`=`)}`);
+
+    return value as T;
+}
+
+export function hasFlag(flag: string): boolean {
+    return findIndexFlag(flag) !== -1;
+}
+
+export function findIndexFlag(flag: string): number {
+    return process.argv.findIndex(arg => arg === flag || arg.split(`=`)[0] === flag);
+}
+
+export function stringifier(value?: string): string | undefined {
+    return value === `undefined` || value === `null` ? undefined : value;
+}

--- a/scripts/helpers/colored-log.ts
+++ b/scripts/helpers/colored-log.ts
@@ -1,0 +1,19 @@
+export function processLog(message: string): void {
+    console.info('\x1b[36m%s\x1b[0m', message);
+}
+
+export function errorLog(message: string): void {
+    console.info('\x1B[31m%s\x1B[0m', message);
+}
+
+export function successLog(message: string): void {
+    console.info('\x1B[32m%s\x1B[0m', message);
+}
+
+export function infoLog(message: string): void {
+    console.info('\x1B[34m%s\x1B[0m', message);
+}
+
+export function titleLog(message: string): void {
+    console.info('\x1b[35m', message);
+}

--- a/scripts/helpers/get-all-versions.ts
+++ b/scripts/helpers/get-all-versions.ts
@@ -1,0 +1,7 @@
+import {execSync} from 'child_process';
+
+export function getAllVersions(name: string): string[] {
+    return JSON.parse(
+        execSync(`npm view ${name} versions --json || echo "[]"`).toString(),
+    );
+}

--- a/scripts/helpers/get-last-major-version.ts
+++ b/scripts/helpers/get-last-major-version.ts
@@ -1,0 +1,3 @@
+export function getLastMajorVersion(versions: string[], currentMajor: number): number {
+    return Math.max(...versions.map(x => parseInt(x)), currentMajor);
+}

--- a/scripts/npm-publish.ts
+++ b/scripts/npm-publish.ts
@@ -1,0 +1,41 @@
+import {execSync} from 'child_process';
+import {resolve} from 'path';
+
+import {getValueByFlag} from './helpers/argv';
+import {errorLog, infoLog, processLog, successLog} from './helpers/colored-log';
+import {getAllVersions} from './helpers/get-all-versions';
+import {getLastMajorVersion} from './helpers/get-last-major-version';
+
+const isDryRun =
+    getValueByFlag<'true' | 'false' | 'undefined'>(`--dry-run`, `false`) === `true`;
+const path = getValueByFlag<string>(`--path`, ``);
+
+(async function main(): Promise<void> {
+    const packageJson = await import(resolve(path, `package.json`));
+    const versions: string[] = getAllVersions(packageJson.name);
+
+    if (versions.includes(packageJson.version) && !isDryRun) {
+        errorLog(`${packageJson.name}@${packageJson.version} is already published`);
+
+        return;
+    }
+
+    infoLog(`name: ${packageJson.name}`);
+    infoLog(`version: ${packageJson.version}`);
+
+    const dry = isDryRun ? `--dry-run` : ``;
+    const tag = makeTag(packageJson.version, versions);
+    const command = `npm publish ${path} ${tag} ${dry} --access public`;
+
+    processLog(command);
+    execSync(command, {stdio: `inherit`, encoding: `utf8`});
+    successLog(`+${packageJson.name}@${packageJson.version} is published successfully`);
+})();
+
+function makeTag(version: string, versions: string[]): string {
+    const currentMajor = parseInt(version);
+    const maxMajorVersion = getLastMajorVersion(versions, currentMajor);
+    const tagFlag = maxMajorVersion > currentMajor ? `--tag v${currentMajor}-lts` : ``;
+
+    return version.includes(`rc`) ? `--tag next` : tagFlag;
+}

--- a/tsconfig.eslint.json
+++ b/tsconfig.eslint.json
@@ -6,6 +6,6 @@
         "strict": false,
         "incremental": true
     },
-    "include": ["projects"],
+    "include": ["projects", "scripts"],
     "exclude": ["**/node_modules", "**/schematics/**", "**/.*/", "*.js"]
 }

